File: backend/app/models/technician_activity_log.py
from sqlalchemy import Column, Integer, String, ForeignKey, DateTime
from sqlalchemy.sql import func
from app.database import Base


class TechnicianActivityLog(Base):
    __tablename__ = "technician_activity_logs"

    id = Column(Integer, primary_key=True, index=True)

    technician_id = Column(
        Integer,
        ForeignKey("users.id"),
        nullable=False,
    )

    service_id = Column(
        Integer,
        ForeignKey("service_history.id", ondelete="CASCADE"),
        nullable=False,
    )

    action = Column(
        String,
        nullable=False,
    )
    # Examples:
    # ASSIGNED
    # STARTED
    # COMPLETED

    created_at = Column(
        DateTime(timezone=True),
        server_default=func.now(),
        nullable=False,
    )

File: mobile/spag_app/lib/features/admin/models/technician_activity_log.dart
class TechnicianActivityLog {
  final int id;
  final int technicianId;
  final int? serviceId;
  final String action;
  final String createdAt;

  TechnicianActivityLog({
    required this.id,
    required this.technicianId,
    this.serviceId,
    required this.action,
    required this.createdAt,
  });

  factory TechnicianActivityLog.fromJson(Map<String, dynamic> json) {
    return TechnicianActivityLog(
      id: json['id'] as int,
      technicianId: json['technician_id'] as int,
      serviceId: json['service_id'] as int?,
      action: json['action'] as String? ?? '',
      createdAt: json['created_at']?.toString() ?? '',
    );
  }
}
